src/core/storage/sqliteStorage/migrations/version-0.0.2.ts

```diff
@@ -0,0 +1,23 @@
+const DATA_V002 = {
+  version: "0.0.2",
+  sql: [
+    `UPDATE items SET id = REPLACE(id, 'metadata:', '');
+    UPDATE items SET name = REPLACE(name, 'metadata:', '');
+    UPDATE items
+      SET value = json_set(
+          value,
+          '$.id',
+          replace(json_extract(value, '$.id'), 'metadata:', '')
+      )
+      WHERE json_extract(value, '$.id') LIKE 'metadata:%';
+    UPDATE items
+      SET value = json_set(
+          value,
+          '$._tags.id',
+          replace(json_extract(value, '$._tags.id'), 'metadata:', '')
+      )
+      WHERE json_extract(value, '$._tags.id') LIKE 'metadata:%';
```

Comment:
Tags at the DB layer have a different format - so this didn't work for the tags. The effect is that if I send an IPEX apply, I can no longer present a valid credential because it searches tags by ID.

```
got item by id EGNUaig8_A3tO72bq76UNxEdjfNsuH7Va_FQ4ue8F7R6: {
  "values": [
    {
      "category": "CredentialMetadataRecord",
      "name": "EGNUaig8_A3tO72bq76UNxEdjfNsuH7Va_FQ4ue8F7R6",
      "value": "{\"_tags\":{\"isArchived\":false,\"isDeleted\":false,\"connectionId\":\"EHZ6H93014ZnZSRmzjqOL3Gn56MNkMydIy-hFnu3hJeX\",\"id\":\"EGNUaig8_A3tO72bq76UNxEdjfNsuH7Va_FQ4ue8F7R6\"},\"type\":\"CredentialMetadataRecord\",\"id\":\"EGNUaig8_A3tO72bq76UNxEdjfNsuH7Va_FQ4ue8F7R6\",\"isArchived\":false,\"isDeleted\":false,\"createdAt\":\"2024-07-26T08:47:31.266Z\",\"issuanceDate\":\"2024-07-26T08:47:18.415Z\",\"credentialType\":\"IIW 2024 Demo Day Attendee\",\"status\":\"confirmed\",\"connectionId\":\"EHZ6H93014ZnZSRmzjqOL3Gn56MNkMydIy-hFnu3hJeX\",\"updatedAt\":\"2024-07-26T08:47:32.521Z\"}",
      "tags": "boolean|isArchived|0,boolean|isDeleted|0,string|connectionId|EHZ6H93014ZnZSRmzjqOL3Gn56MNkMydIy-hFnu3hJeX,string|id|metadata:EGNUaig8_A3tO72bq76UNxEdjfNsuH7Va_FQ4ue8F7R6"
    }
  ]
}
```

Reply:
@iFergal I pushed, but there were some minor issue:

-  After have migration ts prefix, no need to migrate SQL anymore. Can we remove it?
-  Currently, I write migration ts in `updates` folder and use it at `sqliteSession.ts`. Is it structural confusion between sql migration and ts migration? Should we separate migration ts and call it from outside the `sqliteStorage` directory (from `storage`)?

Reply:
Yes, remove the SQL migration file.

For the structure, I have a question:
- We will use TS migration style for migrating records we create
- SQL is used to init the original database
- But if in the future we wanted to do some normal SQL migration - e.g. add some index or new column etc (unlikely) - how would we mix these 2? Is it possible?

Reply:
Of course, it's okay. In TS migration, we return array contain normal SQL vs values (parameter). If we want execute some normal SQL, we just only push into this array

Reply:
OK, in this case it's all part of the same migrations - sometimes it's SQL and almost always it will be TS.

So I would just keep them all in the main `migrations` directory and name them appropraitely:
- v0.0.1-init_sql.ts
- v0.0.2-credprefix.ts
- ...
- ...
- v1.0.0-newTable_sql.ts

So here they are mixed, and the numbers always increase. We can have `_sql.ts` endings if its SQL, which should just be the init.

Can you add a README that describes the mix of TS and SQL and state that TS should almost always be used?

Reply:
^ and yeah, good practice to include a basic short description in the filename instead of just the version number